webhook/resources/engine/validator.go

```diff
@@ -0,0 +1,44 @@
+package engine
+
+import (
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+	werror "github.com/longhorn/longhorn-manager/webhook/error"
+)
+
+type engineValidator struct {
+	admission.DefaultValidator
+	ds *datastore.DataStore
+}
+
+func NewValidator(ds *datastore.DataStore) admission.Validator {
+	return &engineValidator{ds: ds}
+}
+
+func (e *engineValidator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "engines",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.Engine{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Update,
+		},
+	}
+}
+
+func (e *engineValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	oldEngine := oldObj.(*longhorn.Engine)
+	newEngine := newObj.(*longhorn.Engine)
+
+	if oldEngine.Spec.BlockSize != newEngine.Spec.BlockSize {
+		return werror.NewInvalidError("Changing blockSize is not allowable", "")
+	}
```

Comment:
Can we add an engine mutator to update `spec.blockSize` for existing engines?
Then this validator that prevents the **non-zero** `e.spec.blockSize` from being modified only.

Reply:
Yes, updated. Thank you.